recipe/conda_build_config.yaml

```diff
@@ -5,5 +5,5 @@ mpi:
 
 pin_run_as_build:
   mpich: x.x
-  openmpi: x.x
+  #openmpi: x.x
   msmpi: x.x
```

Comment:
Why is this section in the recipe at all?

Reply:
I am not 100% sure since I wasn't a maintainer until a few days ago, but I thought this is perfectly legal?
https://conda-forge.org/docs/maintainer/knowledge_base.html?highlight=pin_run_as_build#building-mpi-variants

Reply:
The strangest thing to me is that it's totally fine to have `mpich` and `msmpi` here, but not `openmpi`.

Reply:
I tried a commit of an alternative syntax. This feels like a conda-build bug maybe?

Reply:
I'd guess so. With your fix, both `openmpi`'s `max_pin=x.x` and pinned version `4.0` appear in ci support files, but not for other 2 mpi impls. As if somewhere in the toolchain `openmpi` is picked out.

Reply:
This could be due to the appearance of only openmpi in the build section of the recipe compared to the other providers.

Reply:
Ah, it would make sense.

Reply:
You could rerender with that line in the recipe removed to see if it makes a difference.

Reply:
>> This could be due to the appearance of only openmpi in the build section of the recipe compared to the other providers.
>>
> You could rerender with that line in the recipe removed to see if it makes a difference.

Actually this is the correct fix! 🎉 Looks like a bad interaction between Jinja and the selector...Pushing a fix shortly.